### PR TITLE
Add SBOM scan REST API

### PR DESCRIPTION
**Pull request by stonezdj**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [ ] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: api/v2.0/swagger.yaml [hunk 1]
```diff
@@ -991,6 +991,12 @@ paths:
           type: boolean
           required: false
           default: false
+        - name: with_sbom_overview
+          in: query
+          description: Specify whether the generate sbom overview is include inside the returning artifacts
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:996]: Specify whether the SBOM overview is included with the returning artifacts.

File: api/v2.0/swagger.yaml [hunk 2]
```diff
@@ -1096,6 +1102,12 @@ paths:
           type: boolean
           required: false
           default: false
+        - name: with_sbom_overview
+          in: query
+          description: Specify whether the sbom overview is inclued inside the returning artifacts
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:1107]: Specify whether the SBOM overview is included with the returning artifacts.

File: api/v2.0/swagger.yaml (hunk 1 above)
Comment (Vad1mo) [reply api/v2.0/swagger.yaml:996]: It is not clear from the description what this "**overview**" it? Can you elaborate on that? As a consumer of the API, I would assume there is also a more detailed response available.

File: api/v2.0/swagger.yaml [hunk 3]
```diff
@@ -6592,6 +6609,9 @@ definitions:
       scan_overview:
         $ref: '#/definitions/ScanOverview'
         description: The overview of the scan result.
+      sbom_overview:
+        $ref: '#/definitions/SBOMOverview'
+        description: The overview of the generate SBOM
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:6614]: The overview of the generated SBOM.

File: api/v2.0/swagger.yaml [hunk 4]
```diff
@@ -6738,11 +6758,47 @@ definitions:
         type: string
         description: Version of the scanner adapter
         example: "v0.9.1"
+  ScanRequestType:
+    type: object
+    properties:
+      scan_type: 
+        type: string
+        description: 'The scan type for scan requst, it could be vulnerability or sbom'
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:6766]: The scan type for the scan request. Two options are currently supported, `vulnerability? and `sbom`.

File: api/v2.0/swagger.yaml [hunk 5]
```diff
@@ -6738,11 +6758,47 @@ definitions:
...
     additionalProperties:
       $ref: '#/definitions/NativeReportSummary'
+  SBOMOverview:
+    type: object
+    description: 'The generate SBOM overview information'
+    properties:
+      start_time:
+        type: string
+        format: date-time
+        description: 'The start time of the scan process that generating report'
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:6780]: The start time of the scan process that is generating the report.

File: api/v2.0/swagger.yaml [hunk 6]
```diff
@@ -6738,11 +6758,47 @@ definitions:
...
+        example: '2006-01-02T14:04:05Z'
+      end_time:
+        type: string
+        format: date-time
+        description: 'The end time of the scan process that generating report'
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:6785]: The end time of the scan process that is generating the report.

File: api/v2.0/swagger.yaml [hunk 7]
```diff
@@ -6738,11 +6758,47 @@ definitions:
...
+        example: '2006-01-02T15:04:05Z'
+      scan_status:
+        type: string
+        description: 'The status of the generate SBOM process'
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:6789]: The process status for the generated SBOM. 
> Here it is not clear what does the status report? Fail Success or more? What can the user find in the satus?

File: api/v2.0/swagger.yaml [hunk 8]
```diff
@@ -6738,11 +6758,47 @@ definitions:
...
+        type: string
+        description: 'The digest of the generated SBOM accessory'   
+      report_id:
+        type: string
+        description: 'id of the native scan report'
+        example: '5f62c830-f996-11e9-957f-0242c0a89008'   
+      duration: 
+        type: integer
+        format: int64
+        description: 'The seconds spent for generating the report'
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:6800]: Time in seconds required to create the report.

File: api/v2.0/swagger.yaml [hunk 9]
```diff
@@ -8368,7 +8424,12 @@ definitions:
         default: ""
         description: Indicate the healthy of the registration
         example: "healthy"
-
+      capabilities:
+        type: object
+        description: Indicate the scanner's properties such as support_vulnerability or support_sbom
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:8429]: Indicates the capabilities of the scanner, e.g. support_vulnerability or support_sbom.

**Review by Vad1mo — CHANGES_REQUESTED** [review]

**Review by wy65701436 — APPROVED** [review]
lgtm

File: api/v2.0/swagger.yaml [hunk 10]
```diff
@@ -6738,11 +6758,47 @@ definitions:
         type: string
         description: Version of the scanner adapter
         example: "v0.9.1"
+  ScanRequestType:
+    type: object
+    properties:
+      scan_type: 
+        type: string
```
Comment (zyyw) [inline api/v2.0/swagger.yaml:6765]: should this be an enum?

**Review by zyyw — APPROVED** [review]

**Pull request merged**